data-analytics/cloud-composer-examples/composer_dataflow_examples/dataflow/process_delimited.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""process_delimited.py is a Dataflow pipeline which reads a delimited file,
+adds some additional metadata fields and loads the contents to a BigQuery table."""
+
+from __future__ import absolute_import
+import argparse
+import logging
+import ntpath
+import re
+
+import apache_beam as beam
+from apache_beam.options import pipeline_options
+
+
+class RowTransformer(object):
+    """A helper class that contains utility methods to parse the delimited file
+    and convert every record into a format acceptable to BigQuery. It also contains
+    utility methods to add a load_dt and a filename fields as demonstration of
+    how records can be enriched as part of the load process."""
+
+    def __init__(self, delimiter, header, filename, load_dt):
+        self.delimiter = delimiter
+
+        # Extract the field name keys from the comma separated input.
+        self.keys = re.split(',', header)
```

Comment:
Is the header always comma-separated, even when the delimiter is not?

Reply:
That is correct. The header in the example is always a comma separated input added as an Airflow variable. I have described that in the README.